.github/workflows/ci.yml

```
@@ -25,5 +25,14 @@ jobs:
         version: latest
         skip-cache: true
 
-    - name: Run Go tests
-      run: go test -v -race ./...
+    - name: Run Go tests and generate coverage report
+      run: go test -v -race ./... -coverprofile=coverage.out
+    - name: Upload the coverage report
+      uses: actions/upload-artifact@65462800fd760344b1a7b4382951275a0abb4808
+      with:
+         path: coverage.out
+         name: Coverage-report
+    - name: Display Coverage Report 
```

Same comment as above
```suggestion

    - name: Display Coverage Report 
```